lib/floe/workflow/states/map.rb

```diff
@@ -58,8 +60,13 @@ def start(context)
         end
 
         def finish(context)
-          result = context.state["ItemProcessorContext"].map { |ctx| Context.new(ctx).output }
-          context.output = process_output(context, result)
+          if failed?(context)
+            error = parse_error(context)
+            retry_state!(context, error) || catch_error!(context, error) || fail_workflow!(context, error)
+          else
+            result = each_item_processor(context).map(&:output)
+            context.output = process_output(context, result)
+          end
```

Comment:
Please change to `success?()` to mirror `Task`

Reply:
+1 for preferring positively toned methods (avoids double-negatives and such)